plutus-core/untyped-plutus-core/test/DeBruijn/FlatNatWord.hs

```diff
@@ -0,0 +1,51 @@
+module DeBruijn.FlatNatWord (test_flatNatWord) where
+
+import Common
+import Flat
+import Hedgehog
+import Hedgehog.Gen qualified as Gen
+import Hedgehog.Range qualified as Range
+import Numeric.Natural
+import Test.Tasty.Hedgehog
+
+{- NOTE: [DeBruijjn index flat format]
+In the debruijn branch we are transitioning the debruijn index to be a Word instead of Natural, for CEK performance reasons.
+
+This means the types of `Script`,`Program`,`Term` change and so does the flat "interface".
+
+This module tests that although the interface/types change,
+the underlying flat encoding/decoding remains the same,
+so there is no need to issue a new plutus language version increase.
+-}
+
+test_flatNatWord :: TestNested
+test_flatNatWord = testNested "FlatNatWord"
+    [ pure $ testProperty "natEncByteString = wordEncByteString" prop_Enc
+    , pure $ testProperty "roundtripWord" prop_TripWord
+    , pure $ testProperty "roundtripNat" prop_TripNat
+    ]
+
+prop_Enc :: Property
+prop_Enc = property $ do
+    w <- forAll $ Gen.word64 Range.linearBounded
+    let n = fromIntegral w :: Natural
+    flat w === flat n
+
+prop_TripWord :: Property
+prop_TripWord = property $ do
```

Comment:
you might want to use `tripping` from `hedgehog`

Reply:
in fact, you almost certainly do

Reply:
I checked it before, but its type is not what I wanted : https://hackage.haskell.org/package/hedgehog-1.1.1/docs/Hedgehog.html#v:tripping

Reply:
Isn't it? I'd have thought something like this would work:

```
property $ do 
      w <- forAll $ Gen.word64 Range.linearBounded
      tripping w (flat . fromIntegral @Natural) unflat
```

Reply:
I'm wondering about `linearBounded`.  @michaelpj , didn't we have a problem with something like this in the `SatInt` tests where the generator wasn't generating the endpoints, which is where things tend to go wrong?

Reply:
Ah, I found [the relevant tests](https://github.com/input-output-hk/plutus/blob/5ec689b1e4411d3c29f53485e3645b617eabb70b/plutus-core/plutus-core/satint-test/TestSatInt.hs#L79).  That was QuickCheck though.

Reply:
Yeeees, although in this case I think so long as it produces a *couple* of very big values it's fine. If it does `maxBound` ever (which I hope that it will) then we should be fine.

Reply:
good idea, i will add explicitly the extreme cases of minbound and maxbound just to be sure